client/components/data/query-user-purchases/index.jsx

```diff
@@ -29,4 +29,18 @@ function QueryUserPurchases() {
 	return null;
 }
 
+export const useQueryUserPurchases = ( disabled = false ) => {
```

Comment:
This `disabled` flag feels weird. Shouldn't "whether or not to query the data" be the responsibility of the calling component?

Reply:
Unfortunately, we can't. Hooks cannot be conditionally rendered. I don't know if there is a pattern that I'm not aware of.
I'm open to suggestions!

Reply:
@p-jackson Any great suggestions here?

Reply:
I think passing the disabled parameter is a common practice among hooks. The component can avoid making the requests using that parameter.

Reply:
Such hack. So wow.

Reply:
Yeah this is one of the things about hooks; things that you're used to treating as imperative (`if condition is met then query purchases`) have become declarative (`declare a purchase query with property of condition being met`).
This is the ReactQuery does it. [It's often used for waiting for all endpoint params to be available](https://github.com/Automattic/wp-calypso/blob/dd409ed59f831e4a0a1bb59cc8f3a8d3c7703e2a/client/data/site-roles/use-site-roles-query.js#L12).

If you were inside a component you'd be able to do something like `isCondtion && <QueryUserPurchases />`, but since you want to be able to do this inside the `useCopySite()` mega-hook you've got to do it with a hook.

I know I originally suggested the `disabled` flag, but if you wanted to make it look more familiar you could copy the ReactQuery API by flipping it to an `enabled` flag and passing it as part of an options object. That way devs will see it as just another hook with "Query" in the name, and they'll be like: oh yeah I know how this works.

Reply:
Thanks @p-jackson for your suggestion. I just changed `disable` to `enable` to match ReactQuery API.